web3swift/Web3/Classes/Web3+Options.swift

```diff
@@ -81,4 +81,21 @@ public struct Web3Options {
         }
         return newOptions
     }
+    
+    public static func smartMergeGasLimit(originalOptions: Web3Options?, extraOptions: Web3Options?, gasEstimage: BigUInt) -> BigUInt? {
+        guard let mergedOptions = Web3Options.merge(originalOptions, with: extraOptions) else {return nil} //just require any non-nils
+        if mergedOptions.gasLimit == nil {
```

Comment:
too many unnecessary ifs.
`Web3Options.merge` has already done all the work.
You just might return what is greater `mergedOptions.gasLimit` or `gasEstimage`.